MonoGame.Framework/Vector2.cs

```diff
@@ -756,9 +756,11 @@ public static void Negate(ref Vector2 value, out Vector2 result)
         /// </summary>
         public void Normalize()
         {
-            float val = 1.0f / (float)Math.Sqrt((X * X) + (Y * Y));
-            X *= val;
-            Y *= val;
+			var magnitude = (float)Math.Sqrt((X * X) + (Y * Y));
+			if(magnitude > 1E-05f)
```

Comment:
> Is adhering to XNA really a top priority at this point? 

Yes... especially when it comes to math.  The last thing we want is a physics engine behaving differently on MonoGame because we decided to change the behavior that has been the same for almost 10 years now.

> At the very least it would be nice to have a method that "does the right thing" even if it has a slightly odd name. 

Which is what I suggested.

> That coupled with a method summary explaining that NaN is a possible 

Absolutely...  adding more to the reference docs is always welcome as long as it is written correctly.